### ISSUE TITLE
Car not changing states when charging complete.

### ISSUE THREAD

**Issue by drklain** [adriankumpf/teslamate#203]
**Describe the bug**
When the car finishes charging, the state remains "charging" instead of shifting to "online".  State does not appear to change until the car is driven and teslamate detects the car has changed to "driving."

Seems to have come with version 1.11

**Comment by adriankumpf** [comment]
The relevant entries from the logs of the teslamate container are needed to be able to investigate this.

**Comment by drklain** [comment]
My apologies. Have been away on travel. Not sure what logs you need, please let me know and I'll upload them.  I am still seeing this behavior.  Car charge state just reflects as that state from when it starts charging until I drive the car for the first time after charging it...whether I drive it immediately after charging, 15 minutes after it finishes or 10 hours after it finishes...

**Comment by adriankumpf** [comment]
> ...
>
> **Relevant entries from the logs**
>
> _You **have to** provide the logs of the teslamate container / application unless the bug is 
> limited to Grafana only. To fetch the logs of the teslamte container run the following command:_ `docker-compose logs teslamate`
> 
> \```
> paste logs here
> \```
>
> ...

Strange, that is not supposed to happen. Let's take a look at the logs.

**Comment by drklain** [comment]
Teslamate log


date | stream | content
-- | -- | --
2019-10-19 18:50:29 | stdout | 14:50:29.545 car_id=1 [info] Suspending logging
2019-10-19 18:35:27 | stdout | 14:35:27.630 car_id=1 [info] Suspending logging
2019-10-19 18:20:25 | stdout | 14:20:25.132 car_id=1 [info] Suspending logging
2019-10-19 18:05:24 | stdout | 14:05:24.009 car_id=1 [info] Suspending logging
2019-10-19 17:50:22 | stdout | 13:50:22.494 car_id=1 [info] Suspending logging
2019-10-19 17:35:21 | stdout | 13:35:21.443 car_id=1 [info] Suspending logging
2019-10-19 17:20:20 | stdout | 13:20:20.057 car_id=1 [info] Suspending logging
2019-10-19 17:05:19 | stdout | 13:05:19.000 car_id=1 [info] Suspending logging
2019-10-19 16:50:16 | stdout | 12:50:16.614 car_id=1 [info] Suspending logging
2019-10-19 16:20:16 | stdout | 12:20:16.588 car_id=1 [info] Start / :online
2019-10-19 16:20:16 | stdout | 12:20:16.588 car_id=1 [info] Driving / Ended / 4 km – 8 min
2019-10-19 16:12:39 | stdout | 12:12:39.934 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 16:12:39 | stdout | 12:12:39.290 car_id=1 [info] Driving / Start
2019-10-19 16:05:41 | stdout | 12:05:41.067 car_id=1 [info] Start / :online
2019-10-19 16:05:41 | stdout | 12:05:41.067 car_id=1 [info] Driving / Ended / 4 km – 8 min
2019-10-19 15:57:57 | stdout | 11:57:57.767 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 15:57:57 | stdout | 11:57:57.080 car_id=1 [info] Driving / Start
2019-10-19 15:57:44 | stdout | 11:57:44.684 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 15:57:44 | stdout | 11:57:44.631 car_id=1 [info] Start / :online
2019-10-19 15:57:44 | stdout | 11:57:44.631 car_id=1 [info] Charging / Disconnected / 14.42 kWh – 82 min
2019-10-19 15:57:37 | stdout | 11:57:37.586 [info] Sent 204 in 1ms
2019-10-19 15:57:37 | stdout | 11:57:37.586 car_id=1 [info] Resuming logging
2019-10-19 15:57:37 | stdout | 11:57:37.584 [info] PUT /api/car/1/logging/resume
2019-10-19 15:44:16 | stdout | 11:44:16.071 car_id=1 [info] Suspending logging
2019-10-19 15:29:13 | stdout | 11:29:13.060 car_id=1 [info] Suspending logging
2019-10-19 15:14:10 | stdout | 11:14:10.069 car_id=1 [info] Suspending logging
2019-10-19 14:59:07 | stdout | 10:59:07.058 car_id=1 [info] Suspending logging
2019-10-19 14:44:04 | stdout | 10:44:04.062 car_id=1 [info] Suspending logging
2019-10-19 14:29:00 | stdout | 10:29:00.948 car_id=1 [info] Suspending logging
2019-10-19 14:13:57 | stdout | 10:13:57.927 car_id=1 [info] Suspending logging
2019-10-19 13:58:54 | stdout | 09:58:54.975 car_id=1 [info] Suspending logging
2019-10-19 13:43:53 | stdout | 09:43:53.012 car_id=1 [info] Suspending logging
2019-10-19 13:35:35 | stdout | 09:35:35.646 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 13:35:35 | stdout | 09:35:35.455 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 13:35:35 | stdout | 09:35:35.238 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 13:35:34 | stdout | 09:35:34.753 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 13:35:34 | stdout | 09:35:34.701 [info] Adding elevation to 4 positions ...
2019-10-19 13:28:50 | stdout | 09:28:50.848 car_id=1 [info] Suspending logging
2019-10-19 13:13:48 | stdout | 09:13:48.813 car_id=1 [info] Suspending logging
2019-10-19 12:58:46 | stdout | 08:58:46.802 car_id=1 [info] Suspending logging
2019-10-19 12:43:44 | stdout | 08:43:44.780 car_id=1 [info] Suspending logging
2019-10-19 12:28:42 | stdout | 08:28:42.704 car_id=1 [info] Suspending logging
2019-10-19 12:13:40 | stdout | 08:13:40.680 car_id=1 [info] Suspending logging
2019-10-19 11:58:37 | stdout | 07:58:37.661 car_id=1 [info] Suspending logging
2019-10-19 11:28:21 | stdout | 07:28:21.309 car_id=1 [info] Charging / Complete / 14.42 kWh – 82 min
2019-10-19 10:06:10 | stdout | 06:06:10.785 car_id=1 [info] Charging / SOC: 42%
2019-10-19 09:51:08 | stdout | 05:51:08.433 car_id=1 [info] Suspending logging
2019-10-19 09:36:05 | stdout | 05:36:05.433 car_id=1 [info] Suspending logging
2019-10-19 09:21:02 | stdout | 05:21:02.346 car_id=1 [info] Suspending logging
2019-10-19 09:05:59 | stdout | 05:05:59.295 car_id=1 [info] Suspending logging
2019-10-19 08:50:56 | stdout | 04:50:56.406 car_id=1 [info] Suspending logging
2019-10-19 08:35:53 | stdout | 04:35:53.209 car_id=1 [info] Suspending logging
2019-10-19 08:20:50 | stdout | 04:20:50.174 car_id=1 [info] Suspending logging
2019-10-19 08:05:46 | stdout | 04:05:46.493 car_id=1 [info] Suspending logging
2019-10-19 07:35:34 | stdout | 03:35:34.370 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 07:35:34 | stdout | 03:35:34.050 car_id=1 [info] Start / :online
2019-10-19 06:38:54 | stdout | 02:38:54.317 car_id=1 [info] Start / :asleep
2019-10-19 06:23:43 | stdout | 02:23:43.663 car_id=1 [info] Suspending logging
2019-10-19 06:08:41 | stdout | 02:08:41.476 car_id=1 [info] Suspending logging
2019-10-19 05:53:38 | stdout | 01:53:38.184 car_id=1 [info] Suspending logging
2019-10-19 05:42:04 | stdout | 01:42:04.257 [info] Sent 200 in 129ms
2019-10-19 05:42:04 | stdout | 01:42:04.128 [info] GET /
2019-10-19 05:23:23 | stdout | 01:23:23.650 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 05:23:23 | stdout | 01:23:23.417 car_id=1 [info] Start / :online
2019-10-19 04:25:44 | stdout | 00:25:44.557 car_id=1 [info] Start / :asleep
2019-10-19 04:10:33 | stdout | 00:10:33.786 car_id=1 [info] Suspending logging
2019-10-19 03:55:31 | stdout | 23:55:31.857 car_id=1 [info] Suspending logging
2019-10-19 03:40:28 | stdout | 23:40:28.387 car_id=1 [info] Suspending logging
2019-10-19 03:10:23 | stdout | 23:10:23.473 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 03:10:23 | stdout | 23:10:23.275 car_id=1 [info] Start / :online
2019-10-19 02:51:49 | stdout | 22:51:49.978 car_id=1 [info] Start / :asleep
2019-10-19 02:36:39 | stdout | 22:36:39.339 car_id=1 [info] Suspending logging
2019-10-19 02:21:36 | stdout | 22:21:36.356 car_id=1 [info] Suspending logging
2019-10-19 02:06:33 | stdout | 22:06:33.279 car_id=1 [info] Suspending logging
2019-10-19 01:51:30 | stdout | 21:51:30.341 car_id=1 [info] Suspending logging
2019-10-19 01:36:27 | stdout | 21:36:27.261 car_id=1 [info] Suspending logging
2019-10-19 01:21:24 | stdout | 21:21:24.234 car_id=1 [info] Suspending logging
2019-10-19 01:06:21 | stdout | 21:06:21.154 car_id=1 [info] Suspending logging
2019-10-19 00:51:18 | stdout | 20:51:18.119 car_id=1 [info] Suspending logging
2019-10-19 00:36:15 | stdout | 20:36:15.098 car_id=1 [info] Suspending logging
2019-10-19 00:21:10 | stdout | 20:21:10.096 car_id=1 [info] Suspending logging
2019-10-19 00:10:48 | stdout | 20:10:48.491 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 00:10:48 | stdout | 20:10:47.458 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 00:10:46 | stdout | 20:10:46.849 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 00:10:45 | stdout | 20:10:45.857 [info] Querying location for elevation takes longer than 50ms ...
2019-10-19 00:10:45 | stdout | 20:10:45.806 [info] Adding elevation to 5 positions ...
2019-10-19 00:06:07 | stdout | 20:06:07.043 car_id=1 [info] Suspending logging
2019-10-18 23:51:02 | stdout | 19:51:02.993 car_id=1 [info] Suspending logging
2019-10-18 23:35:59 | stdout | 19:35:59.954 car_id=1 [info] Suspending logging
2019-10-18 23:20:57 | stdout | 19:20:57.904 car_id=1 [info] Suspending logging
2019-10-18 23:05:55 | stdout | 19:05:55.918 car_id=1 [info] Suspending logging
2019-10-18 22:50:53 | stdout | 18:50:53.902 car_id=1 [info] Suspending logging
2019-10-18 22:35:51 | stdout | 18:35:51.836 car_id=1 [info] Suspending logging
2019-10-18 22:20:50 | stdout | 18:20:50.042 car_id=1 [info] Suspending logging
2019-10-18 21:50:42 | stdout | 17:50:42.539 [info] Sent 204 in 2ms
2019-10-18 21:50:42 | stdout | 17:50:42.539 car_id=1 [info] Resuming logging
2019-10-18 21:50:42 | stdout | 17:50:42.536 [info] PUT /api/car/1/logging/resume
2019-10-18 21:39:27 | stdout | 17:39:27.794 car_id=1 [info] Suspending logging
2019-10-18 21:24:24 | stdout | 17:24:24.804 car_id=1 [info] Suspending logging
2019-10-18 21:09:21 | stdout | 17:09:21.699 car_id=1 [info] Suspending logging
2019-10-18 20:54:19 | stdout | 16:54:19.616 car_id=1 [info] Suspending logging
2019-10-18 20:39:17 | stdout | 16:39:17.784 car_id=1 [info] Suspending logging
2019-10-18 20:24:15 | stdout | 16:24:15.713 car_id=1 [info] Suspending logging
2019-10-18 20:09:13 | stdout | 16:09:13.756 car_id=1 [info] Suspending logging
2019-10-18 19:54:12 | stdout | 15:54:12.544 car_id=1 [info] Suspending logging
2019-10-18 19:39:11 | stdout | 15:39:11.434 car_id=1 [info] Suspending logging
2019-10-18 19:24:08 | stdout | 15:24:08.012 car_id=1 [info] Suspending logging
2019-10-18 18:51:50 | stdout | 14:51:50.037 car_id=1 [info] Start / :online
2019-10-18 18:49:38 | stdout | 14:49:38.105 car_id=1 [info] Start / :offline
2019-10-18 18:49:27 | stdout | }
2019-10-18 18:49:27 | stdout | "response" => nil
2019-10-18 18:49:27 | stdout | "error_description" => "",
2019-10-18 18:49:27 | stdout | "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
2019-10-18 18:49:27 | stdout | 14:49:27.887 [error] TeslaApi.Error / 408 – %{
2019-10-18 18:49:22 | stdout | 14:49:22.711 car_id=1 [warn] Error / upstream timeout
2019-10-18 18:49:22 | stdout | 14:49:22.711 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-18 18:48:51 | stdout | 14:48:51.603 car_id=1 [info] Start / :online
2019-10-18 18:48:51 | stdout | 14:48:51.603 car_id=1 [info] Driving / Ended / 26 km – 29 min
2019-10-18 18:19:57 | stdout | 14:19:57.149 car_id=1 [info] Driving / Start
2019-10-18 18:19:21 | stdout | 14:19:21.356 [info] Sent 204 in 1ms
2019-10-18 18:19:21 | stdout | 14:19:21.354 [info] PUT /api/car/1/logging/resume
2019-10-18 18:12:06 | stdout | 14:12:06.169 car_id=1 [info] Start / :online
2019-10-18 18:12:06 | stdout | 14:12:06.169 car_id=1 [info] Driving / Ended / 35 km – 23 min
2019-10-18 18:10:45 | stdout | 14:10:45.596 [info] Querying location for elevation takes longer than 50ms ...
2019-10-18 18:09:23 | stdout | 14:09:23.290 car_id=1 [warn] Error / upstream timeout
2019-10-18 18:09:23 | stdout | 14:09:23.290 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-18 17:48:57 | stdout | 13:48:57.681 [info] Querying location for elevation takes longer than 50ms ...
2019-10-18 17:48:54 | stdout | 13:48:54.665 [info] Querying location for elevation takes longer than 50ms ...
2019-10-18 17:48:54 | stdout | 13:48:54.512 car_id=1 [info] Driving / Start
2019-10-18 17:48:48 | stdout | 13:48:48.648 [info] Sent 204 in 2ms
2019-10-18 17:48:48 | stdout | 13:48:48.648 car_id=1 [info] Resuming logging
2019-10-18 17:48:48 | stdout | 13:48:48.646 [info] PUT /api/car/1/logging/resume
2019-10-18 17:38:18 | stdout | 13:38:18.216 car_id=1 [info] Suspending logging
2019-10-18 17:23:15 | stdout | 13:23:15.080 car_id=1 [info] Suspending logging
2019-10-18 17:08:13 | stdout | 13:08:13.070 car_id=1 [info] Suspending logging
2019-10-18 16:53:11 | stdout | 12:53:11.061 car_id=1 [info] Suspending logging
2019-10-18 16:38:08 | stdout | 12:38:08.955 car_id=1 [info] Suspending logging
2019-10-18 16:23:06 | stdout | 12:23:06.975 car_id=1 [info] Suspending logging
2019-10-18 16:08:04 | stdout | 12:08:04.965 car_id=1 [info] Suspending logging
2019-10-18 15:53:00 | stdout | 11:53:00.821 car_id=1 [info] Suspending logging
2019-10-18 15:37:57 | stdout | 11:37:57.829 car_id=1 [info] Suspending logging
2019-10-18 15:22:55 | stdout | 11:22:55.831 car_id=1 [info] Suspending logging
2019-10-18 15:22:48 | stdout | 11:22:48.514 car_id=1 [warn] Error / upstream timeout
2019-10-18 15:22:48 | stdout | 11:22:48.514 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-18 15:22:33 | stdout | 11:22:33.373 car_id=1 [warn] Error / upstream timeout
2019-10-18 15:22:33 | stdout | 11:22:33.373 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-18 15:22:18 | stdout | 11:22:18.157 car_id=1 [warn] Error / upstream timeout
2019-10-18 15:22:18 | stdout | 11:22:18.157 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-18 15:22:03 | stdout | 11:22:03.016 car_id=1 [warn] Error / upstream timeout
2019-10-18 15:22:03 | stdout | 11:22:03.016 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-18 15:21:47 | stdout | 11:21:47.891 car_id=1 [warn] Error / upstream timeout
2019-10-18 15:21:47 | stdout | 11:21:47.890 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-18 15:21:32 | stdout | 11:21:32.741 car_id=1 [warn] Error / upstream timeout
2019-10-18 15:21:32 | stdout | 11:21:32.741 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-18 15:21:17 | stdout | 11:21:17.586 car_id=1 [warn] Error / upstream timeout
2019-10-18 15:21:17 | stdout | 11:21:17.586 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-18 15:21:02 | stdout | 11:21:02.345 car_id=1 [warn] Error / upstream timeout
2019-10-18 15:21:02 | stdout | 11:21:02.345 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-18 14:52:08 | stdout | 10:52:08.212 car_id=1 [info] Start / :online
2019-10-18 14:52:08 | stdout | 10:52:08.212 car_id=1 [info] Driving / Ended / 17 km – 24 min
2019-10-18 14:27:43 | stdout | 10:27:43.333 [info] Querying location for elevation takes longer than 50ms ...
2019-10-18 14:27:42 | stdout | 10:27:42.878 car_id=1 [info] Driving / Start
2019-10-18 14:27:01 | stdout | 10:27:00.794 [info] Sent 204 in 1ms
2019-10-18 14:27:00 | stdout | 10:27:00.792 [info] PUT /api/car/1/logging/resume
2019-10-18 14:26:46 | stdout | 10:26:46.478 [info] Querying location for elevation takes longer than 50ms ...
2019-10-18 14:26:46 | stdout | 10:26:46.425 car_id=1 [info] Start / :online
2019-10-18 14:26:46 | stdout | 10:26:46.425 car_id=1 [info] Charging / Disconnected / 2.21 kWh – 12 min
2019-10-18 14:22:23 | stdout | 10:22:23.909 car_id=1 [warn] Preconditioning prevents car to go to sleep
2019-10-18 14:07:22 | stdout | 10:07:22.653 car_id=1 [info] Suspending logging
2019-10-18 13:52:20 | stdout | 09:52:20.678 car_id=1 [info] Suspending logging
2019-10-18 13:37:17 | stdout | 09:37:17.769 car_id=1 [info] Suspending logging
2019-10-18 13:22:14 | stdout | 09:22:14.614 car_id=1 [info] Suspending logging
2019-10-18 13:07:11 | stdout | 09:07:11.610 car_id=1 [info] Suspending logging
2019-10-18 12:59:52 | stdout | 08:59:52.994 [info] Sent 200 in 89ms
2019-10-18 12:59:52 | stdout | 08:59:52.905 [info] GET /
2019-10-18 12:52:08 | stdout | 08:52:08.503 car_id=1 [info] Suspending logging
2019-10-18 12:37:05 | stdout | 08:37:05.493 car_id=1 [info] Suspending logging
2019-10-18 12:22:02 | stdout | 08:22:02.463 car_id=1 [info] Suspending logging
2019-10-18 12:06:59 | stdout | 08:06:59.554 car_id=1 [info] Suspending logging
2019-10-18 11:51:56 | stdout | 07:51:56.441 car_id=1 [info] Suspending logging
2019-10-18 11:36:53 | stdout | 07:36:53.934 car_id=1 [info] Suspending logging
2019-10-18 11:21:50 | stdout | 07:21:50.314 car_id=1 [info] Suspending logging
2019-10-18 11:06:48 | stdout | 07:06:48.350 car_id=1 [info] Suspending logging
2019-10-18 10:53:13 | stdout | 06:53:13.118 [info] Querying location for elevation takes longer than 50ms ...
2019-10-18 10:53:12 | stdout | 06:53:12.893 [info] Querying location for elevation takes longer than 50ms ...
2019-10-18 10:53:12 | stdout | 06:53:12.404 [info] Querying location for elevation takes longer than 50ms ...
2019-10-18 10:53:12 | stdout | 06:53:12.353 [info] Adding elevation to 4 positions ...
2019-10-18 10:51:45 | stdout | 06:51:45.477 car_id=1 [info] Suspending logging
2019-10-18 10:21:35 | stdout | 06:21:35.868 car_id=1 [info] Charging / Complete / 2.21 kWh – 12 min
2019-10-18 10:09:17 | stdout | 06:09:17.231 car_id=1 [info] Charging / SOC: 57%
2019-10-18 09:54:14 | stdout | 05:54:14.185 car_id=1 [info] Suspending logging
2019-10-18 09:39:13 | stdout | 05:39:13.194 car_id=1 [info] Suspending logging
2019-10-18 09:24:09 | stdout | 05:24:09.121 car_id=1 [info] Suspending logging
2019-10-18 09:09:07 | stdout | 05:09:07.105 car_id=1 [info] Suspending logging
2019-10-18 08:54:05 | stdout | 04:54:05.030 car_id=1 [info] Suspending logging
2019-10-18 08:39:02 | stdout | 04:39:02.458 car_id=1 [info] Suspending logging
2019-10-18 08:23:59 | stdout | 04:23:59.996 car_id=1 [info] Suspending logging
2019-10-18 08:08:57 | stdout | 04:08:57.913 car_id=1 [info] Suspending logging
2019-10-18 07:53:55 | stdout | 03:53:55.002 car_id=1 [info] Suspending logging
2019-10-18 07:38:51 | stdout | 03:38:51.904 car_id=1 [info] Suspending logging
2019-10-18 07:23:48 | stdout | 03:23:48.853 car_id=1 [info] Suspending logging
2019-10-18 07:08:45 | stdout | 03:08:45.908 car_id=1 [info] Suspending logging
2019-10-18 06:53:42 | stdout | 02:53:42.779 car_id=1 [info] Suspending logging
2019-10-18 06:38:39 | stdout | 02:38:39.710 car_id=1 [info] Suspending logging
2019-10-18 06:23:36 | stdout | 02:23:36.723 car_id=1 [info] Suspending logging
2019-10-18 06:08:33 | stdout | 02:08:33.631 car_id=1 [info] Suspending logging
2019-10-18 05:53:31 | stdout | 01:53:31.650 car_id=1 [info] Suspending logging
2019-10-18 05:38:28 | stdout | 01:38:28.608 car_id=1 [info] Suspending logging
2019-10-18 05:23:26 | stdout | 01:23:26.558 car_id=1 [info] Suspending logging
2019-10-18 05:10:44 | stdout | 01:10:44.247 car_id=1 [warn] Error / upstream timeout
2019-10-18 05:09:11 | stdout | 01:09:11.133 car_id=1 [error] Error / :connect_timeout
2019-10-18 05:08:58 | stdout | 01:08:58.046 car_id=1 [warn] Error / upstream timeout
2019-10-18 04:53:11 | stdout | 00:53:11.835 [info] Querying location for elevation takes longer than 50ms ...
2019-10-18 04:53:11 | stdout | 00:53:11.730 car_id=1 [info] Start / :online
2019-10-18 03:57:32 | stdout | 23:57:32.505 car_id=1 [info] Start / :asleep
2019-10-18 03:42:21 | stdout | 23:42:21.908 car_id=1 [info] Suspending logging
2019-10-18 03:27:18 | stdout | 23:27:18.872 car_id=1 [info] Suspending logging
2019-10-18 03:12:16 | stdout | 23:12:16.929 car_id=1 [info] Suspending logging
2019-10-18 02:57:15 | stdout | 22:57:15.848 car_id=1 [info] Suspending logging
2019-10-18 02:42:13 | stdout | 22:42:13.521 car_id=1 [info] Suspending logging
2019-10-18 02:27:12 | stdout | 22:27:12.467 car_id=1 [info] Suspending logging
2019-10-18 02:12:11 | stdout | 22:12:11.528 car_id=1 [info] Suspending logging
2019-10-18 01:59:18 | stdout | 21:59:18.706 [info] Sent 200 in 12ms
2019-10-18 01:59:18 | stdout | 21:59:18.694 [info] GET /
2019-10-18 01:57:10 | stdout | 21:57:10.516 car_id=1 [info] Suspending logging
2019-10-18 01:42:09 | stdout | 21:42:09.538 car_id=1 [info] Suspending logging
2019-10-18 01:27:08 | stdout | 21:27:08.435 car_id=1 [info] Suspending logging
2019-10-18 00:57:02 | stdout | 20:57:02.363 [info] Sent 204 in 2ms
2019-10-18 00:57:02 | stdout | 20:57:02.361 [info] PUT /api/car/1/logging/resume
2019-10-18 00:43:50 | stdout | 20:43:50.883 car_id=1 [info] Start / :online
2019-10-18 00:43:50 | stdout | 20:43:50.883 car_id=1 [info] Driving / Ended / 1 km – 2 min
2019-10-18 00:41:32 | stdout | 20:41:32.945 [info] Querying location for elevation takes longer than 50ms ...
2019-10-18 00:41:32 | stdout | 20:41:32.798 car_id=1 [info] Driving / Start
2019-10-18 00:26:43 | stdout | 20:26:43.763 car_id=1 [warn] Preconditioning prevents car to go to sleep
2019-10-18 00:11:42 | stdout | 20:11:42.419 car_id=1 [info] Suspending logging
2019-10-17 23:56:39 | stdout | 19:56:39.470 car_id=1 [info] Suspending logging
2019-10-17 23:41:37 | stdout | 19:41:37.424 car_id=1 [info] Suspending logging
2019-10-17 23:26:35 | stdout | 19:26:35.431 car_id=1 [info] Suspending logging
2019-10-17 23:11:33 | stdout | 19:11:33.423 car_id=1 [info] Suspending logging
2019-10-17 22:56:30 | stdout | 18:56:30.360 car_id=1 [info] Suspending logging
2019-10-17 22:41:29 | stdout | 18:41:29.430 car_id=1 [info] Suspending logging
2019-10-17 22:26:26 | stdout | 18:26:26.284 car_id=1 [info] Suspending logging
2019-10-17 22:11:23 | stdout | 18:11:23.176 car_id=1 [info] Suspending logging
2019-10-17 21:56:20 | stdout | 17:56:20.150 car_id=1 [info] Suspending logging
2019-10-17 21:41:17 | stdout | 17:41:17.158 car_id=1 [info] Suspending logging
2019-10-17 21:26:14 | stdout | 17:26:14.145 car_id=1 [info] Suspending logging
2019-10-17 21:11:10 | stdout | 17:11:10.978 car_id=1 [info] Suspending logging
2019-10-17 20:41:09 | stdout | 16:41:09.264 [info] Querying location for elevation takes longer than 50ms ...
2019-10-17 20:41:09 | stdout | 16:41:09.212 car_id=1 [info] Start / :online
2019-10-17 20:41:09 | stdout | 16:41:09.212 car_id=1 [info] Charging / Stopped / 8.100000000000001 kWh – 47 min
2019-10-17 20:26:05 | stdout | 16:26:05.130 car_id=1 [info] Suspending logging
2019-10-17 20:11:02 | stdout | 16:11:02.962 car_id=1 [info] Suspending logging
2019-10-17 19:56:00 | stdout | 15:56:00.864 car_id=1 [info] Suspending logging
2019-10-17 19:40:58 | stdout | 15:40:58.891 car_id=1 [info] Suspending logging
2019-10-17 19:25:56 | stdout | 15:25:56.827 car_id=1 [info] Suspending logging
2019-10-17 19:10:54 | stdout | 15:10:54.858 car_id=1 [info] Suspending logging
2019-10-17 18:55:52 | stdout | 14:55:52.293 car_id=1 [info] Suspending logging
2019-10-17 18:40:49 | stdout | 14:40:49.861 car_id=1 [info] Suspending logging
2019-10-17 18:25:47 | stdout | 14:25:47.720 car_id=1 [info] Suspending logging
2019-10-17 18:10:45 | stdout | 14:10:45.783 car_id=1 [info] Suspending logging
2019-10-17 17:55:43 | stdout | 13:55:43.891 car_id=1 [info] Suspending logging
2019-10-17 17:40:42 | stdout | 13:40:42.775 car_id=1 [info] Suspending logging
2019-10-17 17:25:39 | stdout | 13:25:39.796 car_id=1 [info] Suspending logging
2019-10-17 17:10:36 | stdout | 13:10:36.621 car_id=1 [info] Suspending logging
2019-10-17 16:55:33 | stdout | 12:55:33.770 car_id=1 [info] Suspending logging
2019-10-17 16:40:30 | stdout | 12:40:30.552 car_id=1 [info] Suspending logging
2019-10-17 16:25:27 | stdout | 12:25:27.609 car_id=1 [info] Suspending logging
2019-10-17 16:10:24 | stdout | 12:10:24.447 car_id=1 [info] Suspending logging
2019-10-17 15:55:21 | stdout | 11:55:21.424 car_id=1 [info] Suspending logging
2019-10-17 15:40:18 | stdout | 11:40:18.415 car_id=1 [info] Suspending logging
2019-10-17 15:25:14 | stdout | 11:25:14.510 car_id=1 [info] Suspending logging
2019-10-17 15:10:12 | stdout | 11:10:12.454 car_id=1 [info] Suspending logging
2019-10-17 14:55:09 | stdout | 10:55:09.322 car_id=1 [info] Suspending logging
2019-10-17 14:40:06 | stdout | 10:40:06.325 car_id=1 [info] Suspending logging
2019-10-17 14:25:04 | stdout | 10:25:04.386 car_id=1 [info] Suspending logging
2019-10-17 14:10:01 | stdout | 10:10:01.242 car_id=1 [info] Suspending logging
2019-10-17 13:54:59 | stdout | 09:54:59.279 car_id=1 [info] Suspending logging
2019-10-17 13:39:57 | stdout | 09:39:57.246 car_id=1 [info] Suspending logging
2019-10-17 13:24:55 | stdout | 09:24:55.063 car_id=1 [info] Suspending logging
2019-10-17 13:09:53 | stdout | 09:09:53.149 car_id=1 [info] Suspending logging
2019-10-17 12:54:50 | stdout | 08:54:50.155 car_id=1 [info] Suspending logging
2019-10-17 12:39:46 | stdout | 08:39:46.998 car_id=1 [info] Suspending logging
2019-10-17 12:24:44 | stdout | 08:24:44.125 car_id=1 [info] Suspending logging
2019-10-17 12:09:40 | stdout | 08:09:40.938 car_id=1 [info] Suspending logging
2019-10-17 11:54:37 | stdout | 07:54:37.993 car_id=1 [info] Suspending logging
2019-10-17 11:39:34 | stdout | 07:39:34.964 car_id=1 [info] Suspending logging
2019-10-17 11:24:30 | stdout | 07:24:30.982 car_id=1 [info] Suspending logging
2019-10-17 10:54:29 | stdout | 06:54:29.026 car_id=1 [info] Charging / Complete / 8.1 kWh – 47 min
2019-10-17 10:47:11 | stdout | 06:47:11.370 [info] Adding elevation to 1 positions ...
2019-10-17 10:07:54 | stdout | 06:07:54.733 car_id=1 [info] Charging / SOC: 50%
2019-10-17 09:52:51 | stdout | 05:52:51.666 car_id=1 [info] Suspending logging
2019-10-17 09:37:48 | stdout | 05:37:48.568 car_id=1 [info] Suspending logging
2019-10-17 09:22:45 | stdout | 05:22:45.730 car_id=1 [info] Suspending logging
2019-10-17 09:07:43 | stdout | 05:07:43.553 car_id=1 [info] Suspending logging
2019-10-17 08:52:41 | stdout | 04:52:41.485 car_id=1 [info] Suspending logging
2019-10-17 08:37:39 | stdout | 04:37:39.951 car_id=1 [info] Suspending logging
2019-10-17 08:22:37 | stdout | 04:22:37.695 car_id=1 [info] Suspending logging
2019-10-17 08:07:35 | stdout | 04:07:35.169 car_id=1 [info] Suspending logging
2019-10-17 07:37:32 | stdout | 03:37:32.399 car_id=1 [info] Start / :online
2019-10-17 06:40:49 | stdout | 02:40:49.651 car_id=1 [info] Start / :asleep
2019-10-17 06:25:38 | stdout | 02:25:38.958 car_id=1 [info] Suspending logging
2019-10-17 06:10:36 | stdout | 02:10:36.826 car_id=1 [info] Suspending logging
2019-10-17 05:55:33 | stdout | 01:55:33.341 car_id=1 [info] Suspending logging
2019-10-17 05:25:24 | stdout | 01:25:24.642 car_id=1 [info] Start / :online
2019-10-17 04:25:45 | stdout | 00:25:45.799 car_id=1 [info] Start / :asleep
2019-10-17 04:10:35 | stdout | 00:10:35.125 car_id=1 [info] Suspending logging
2019-10-17 03:55:32 | stdout | 23:55:32.189 car_id=1 [info] Suspending logging
2019-10-17 03:40:28 | stdout | 23:40:28.719 car_id=1 [info] Suspending logging
2019-10-17 03:10:21 | stdout | 23:10:21.877 car_id=1 [info] Start / :online
2019-10-17 02:11:02 | stdout | 22:11:02.315 car_id=1 [info] Start / :offline
2019-10-17 02:10:46 | stdout | 22:10:46.933 car_id=1 [warn] Error / upstream timeout
2019-10-17 02:10:46 | stdout | 22:10:46.933 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-17 02:10:31 | stdout | 22:10:31.591 car_id=1 [warn] Error / upstream timeout
2019-10-17 02:10:31 | stdout | 22:10:31.591 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-17 02:10:16 | stdout | 22:10:16.191 car_id=1 [warn] Error / upstream timeout
2019-10-17 02:10:16 | stdout | 22:10:16.191 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-17 02:10:00 | stdout | 22:10:00.951 car_id=1 [warn] Error / upstream timeout
2019-10-17 02:10:00 | stdout | 22:10:00.951 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-17 02:09:45 | stdout | 22:09:45.693 car_id=1 [warn] Error / upstream timeout
2019-10-17 02:09:45 | stdout | 22:09:45.693 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-17 02:09:30 | stdout | 22:09:30.482 car_id=1 [warn] Error / upstream timeout
2019-10-17 02:09:30 | stdout | 22:09:30.481 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-17 02:09:15 | stdout | 22:09:15.239 car_id=1 [warn] Error / upstream timeout
2019-10-17 02:09:15 | stdout | 22:09:15.239 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
2019-10-17 01:54:04 | stdout | 21:54:04.551 car_id=1 [info] Suspending logging
2019-10-17 01:39:01 | stdout | 21:39:01.403 car_id=1 [info] Suspending logging
2019-10-17 01:24:00 | stdout | 21:24:00.501 car_id=1 [info] Suspending logging
2019-10-17 01:08:59 | stdout | 21:08:59.501 car_id=1 [info] Suspending logging
2019-10-17 00:53:58 | stdout | 20:53:58.404 car_id=1 [info] Suspending logging
2019-10-17 00:38:56 | stdout | 20:38:56.000 car_id=1 [info] Suspending logging
2019-10-17 00:23:54 | stdout | 20:23:54.671 car_id=1 [info] Suspending logging
2019-10-17 00:08:53 | stdout | 20:08:53.090 car_id=1 [info] Suspending logging

**Comment by adriankumpf** [comment]
So you charge at home and unplug before you leave? Once charging is complete, "charging complete" is displayed Until the charging session is stopped or the charger is disconnected. Afterwards the car state changes to "online" which is also confirmed by the log. That has always been the case and has not changed with v1.11 (or previous releases). Am I missing something or you mean something else?

**Comment by drklain** [comment]
I do charge at home and it is consistent.  The issue (or change in behavior since 1.11) is that it used to change from yellow “charging” to blue “online” on the states graph once charging ended and would be blue until I drove and then it would show purple “driving.”  Now it stays ‘yellow” even though charging is over and remains that way until I take the car for a drive…it doesn’t turn back “blue” when charging is complete/stops…

**Comment by adriankumpf** [comment]
Closing in favor of #211
